src/GraphQL/Execution/ParallelExecutionStrategy.cs

```diff
@@ -44,5 +44,19 @@ await OnBeforeExecutionStepAwaitedAsync(context)
                 pendingNodes.AddRange(childNodes);
             }
         }
+
+        protected override async Task<ExecutionNode> ExecuteNodeAsync(ExecutionContext context, ExecutionNode node)
+        {
+            if (context.Services == null)
+            {
+                return await ExecuteNodeAsync(context, node, null)
+                    .ConfigureAwait(false);
+            }
+            using (var serviceProvider = new ServiceProviderProxy(context.Services))
```

Comment:
So you create a new scope on each execute iteration. This may not be what the client code expects. If you take an ASP.NET Core, then a scope is created on each HTTP request and dependencies are resolved in it. Here you create an indefinite number of nested scopes in which dependencies will be resolved in an independent manner. DbContext (as an example) will be created many times - depending on the number of fields that the client defined in the request.

Did I understand your intentions correctly?

Reply:
Yes, those are my exact intentions.  For a use case sample, take a look at the open source GraphQL.EntityFramework library by Simon Cropp.  The "root" resolve function gets all of the data, and all the child resolve functions do not need access to the dbcontext.  Hence why the ServiceProviderProxy exists - because most resolve functions do not need access to the service provider.  (Understandably, this use case is somewhat specific to those that use EF.)  However, other queries can and should execute in parallel for optimal speed - which requires a separate dbcontext.

Reply:
Since almost no classes support multithreading, any code running under a ParallelExecutionStrategy would have to have another scope for each iteration.  As you say, this may be unexpected, in which case SerialExecutionStrategy should really be the default (when using scoped services).  But that's just a preference, I suppose, as REST adapters would probably want a parallel strategy by default.

Reply:
Well, at least now the context and purpose are clear.